tests/compiler/test_typing.py

```diff
@@ -643,18 +588,23 @@ def test_float_type_argument_plugin(snippetcompiler, caplog):
     assert Integer().validate(x)
 
 
-def test_float_type_argument_plugin_error(snippetcompiler, caplog):
-    snippetcompiler.setup_for_error(
-        """
+@pytest.mark.parametrize("val", [42, "test"])
+def test_float_type_argument_plugin_error(snippetcompiler, val):
+    snippet = f"""
 import test_674
 
-test = test_674::test_float_to_int("test")
-        """,
-        ("Invalid value 'test', expected float (reported in " "test_674::test_float_to_int('test') ({dir}/main.cf:4))"),
+test = test_674::test_float_to_int({val if val == 42 else '"'+val+'"'})
```

Comment:
I think it reads nicer by calling `repr` which adds the single quotes for strings: e.g.

```py
    snippet = f"""
import test_674

test = test_674::test_float_to_int({repr(val)})
        """
    msg = f"Invalid value '{val}', expected float (reported in test_674::test_float_to_int({repr(val)}) ({{dir}}/main.cf:4))"
```

Reply:
Ho I didn't know that, I'll try it